pxtlib/blocks.ts

```diff
@@ -82,15 +82,15 @@ namespace pxt.blocks {
         if (!b) return b;
         // normalize and validate common errors
         // made while translating
-        let nb = b.replace(/[^\\]%\s+/g, '%');
+        let nb = b.replace(/(?:^|[^\\])([%$])\s+/g, '$1');
```

Comment:
two fixes; make this also check blocks using the $ format, and add a check to also fix blocks that start with a parameter